src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -121,7 +121,10 @@ std::vector<Uptane::Target> SotaUptaneClient::findForEcu(const std::vector<Uptan
 data::InstallationResult SotaUptaneClient::PackageInstall(const Uptane::Target &target) {
   LOG_INFO << "Installing package using " << package_manager_->name() << " package manager";
   try {
-    return package_manager_->install(target);
+    // The DockerAppManager needs the actual Target as defined targets.json so
+    // it can access the custom data.
+    auto repo_target = findTargetInDelegationTree(target);
+    return package_manager_->install(*repo_target);
```

Comment:
Why is the target received via `uptaneInstall` not good enough? It's whatever you provide it, normally what you receive from `checkUpdates`. Is it an issue of the version from Director vs. Images? If so, this is another "known issue" type of thing. I'm trying to think of better logic to handle comparing and synchronizing the metadata.

Reply:
Its a "Director's" version of the target. What you get from the Director will not have the custom data in it. So things worked just fine when I first tested things with aktualizr-lite, but when I started testing with OTA connect it failed.

Reply:
Yeah, I'm not surprised it didn't work. Unfortunately that won't "just work". What really needs to happen is that we need logic to determine which things must match and which things can be pulled from one or the other. The first part is partly related to an Uptane question that was resolved in https://github.com/uptane/uptane-standard/pull/98, which we haven't fully implemented yet. The second part is up to the implementation to decide. I don't think we can get away with simply throwing away what is in unique to the Director version, and clearly just ignoring what is in the Images version doesn't work for you. I'm aware that we need to sort this out, but it's not even on my roadmap right now. You could try to take a stab at it (or just simply use the Director version and copy the custom field in question from the Images version and leave the rest of this issue untouched) or you can wait until someone on our side can get around to it.

Reply:
Now that i've gotten most of the target iterator stuff out of sotauptaneclient, I think I could handle this look up directly in the docker-app logic so that we aren't throwing away the original director target. I'll give that a show and see what happens